### PR TITLE
deprecate melee damage enchantments

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
To allow using of json damages, and to get rid of hardcoded json damages
#### Describe the solution
Replace ITEM_DAMAGE_BASH and similar enchantments with field, that does the same
Replace all json that used old values
Handle errors, for they are expected
#### Testing
Loaded a world with item, that had deprecated enchantment value, the game loaded fine, item still deals increased damage
#### Additional context
Rest in peace melee weapon enchantments for dreamsmith, for i don't know how to replace you in artifact code

### CODE REVIEW

**Comment by RedMisao** [comment]
Agree with the template idea. An example should be present at MAGIC.md, in the example eocs file, or referencing an existing use case. Otherwise it's quite difficult to grasp how it can be used.

**Comment by GuardianDll** [comment]
i meant c++ template xd
what do you mean? i think i documented it, no?

File: src/magic_enchantment.cpp [hunk 1]
```diff
@@ -504,8 +529,10 @@ void enchant_cache::load( const JsonObject &jo, const std::string_view,
                     values_multiply.emplace( value, mult );
                 }
             } catch( ... ) {
-                debugmsg( "A relic attempted to load invalid enchantment %s.  If you updated versions this may be a removed enchantment and will fix itself.",
-                          value_obj.get_string( "value", "" ) );
+                if( legacy_values.find( value_obj.get_string( "value", "" ) ) == legacy_values.end() ) {
+                    debugmsg( "A relic attempted to load invalid enchantment %s.", value_obj.get_string( "value",
+                              "" ) );
```
Comment (emixa-d) [inline src/magic_enchantment.cpp:534]: Is removing the second sentence of the message intentional, or an accident?

Comment (GuardianDll) [reply src/magic_enchantment.cpp:534]: it is intentional, we do not need to say "well maybe it was intentional" if we have a list of enchantments that were intentionally removed

**Comment by Maleclypse** [comment]
So am I understanding correctly that this will allow someone to add melee_damage bonus damage in mod damage types? Or does this only apply to vanilla damage types?

File: data/mods/Xedra_Evolved/procgen/dreamsmith_procgen.json [hunk 2]
```diff
@@ -75,96 +75,6 @@
         "increment": 1,
         "power_per_increment": 1,
         "ench_has": "WIELD"
-      },
-      {
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/procgen/dreamsmith_procgen.json:79]: What I'll probably do here is make copy-from's of each of the dreamsmith weapons and then see about adding an eoc inside the recipe that selects "of burning" "glistening (acid) etc. So now all (most?) dreamsmith weapons will have something damage wise and it won't cost power points.

**Comment by GuardianDll** [comment]
@maleclypse correct, it supports any damage types, including one defined in mods

File: src/magic_enchantment.cpp (hunk 1 above)
Comment (emixa-d) [reply src/magic_enchantment.cpp:534]: This list is incomplete, see e.g. https://github.com/CleverRaven/Cataclysm-DDA/pull/63629/files. For example, `ITEM_ENCUMBRANCE` is missing.

Comment (GuardianDll) [reply src/magic_enchantment.cpp:534]: It was pre 0.G, we do not support game migrations more than 1 version (ie at this moment, being in the middle of 0.I, only migration from 0.H to 0.I is supported)

File: data/mods/Xedra_Evolved/procgen/dreamsmith_procgen.json (hunk 2 above)
Comment (GuardianDll) [reply data/mods/Xedra_Evolved/procgen/dreamsmith_procgen.json:79]: I don't think it's necessary, there are still plenty of enchantments left that are still useful

**Pull request merged**